modules/packages/HDF5.chpl

```diff
@@ -3871,132 +3871,159 @@ module HDF5 {
-      if memDataspace == FAIL then
-        halt("failed to create memory dataspace");
+        var sid = H5Screate_simple(A.rank, dims, nil);
+        if sid == FAIL then
+          halt("failed to create dataspace");
 
-      // set up the transfer properties list
-      var xferPlist = H5Pcreate(H5P_DATASET_XFER);
-      if xferPlist == FAIL then
-        halt("failed to create transfer properties list");
+        var dataset = H5Dcreate1(fid, dsetName.c_str(), hdf5Type,
+                                 sid, H5P_DEFAULT);
+        if dataset == FAIL then
+          halt("failed to create dataset");
 
-      ret = H5Pset_dxpl_mpio(xferPlist, H5FD_MPIO_COLLECTIVE);
-      if ret == FAIL then
-        halt("failed to set data transfer property list");
+        // create a file dataspace
+        var fileDataspace = H5Dget_space(dataset);
+        if fileDataspace == FAIL then
+          halt("failed to create file dataspace");
 
-      // write data
-      ret = H5Dwrite(dataset, hdf5Type, memDataspace, fileDataspace,
-                     xferPlist, c_ptrTo(A._value.myLocArr!.myElems));
-      if ret == FAIL then
-        halt("failed writing array data to file");
+        var stride: c_array(uint, A.rank);
+        var count: c_array(uint, A.rank);
+        var start: c_array(uint, A.rank);
 
-      // release temporary handles
-      H5Sclose(fileDataspace);
-      H5Sclose(memDataspace);
-      H5Pclose(xferPlist);
+        for i in 0..#A.rank {
+          stride[i] = 1;
+          count[i] = locDom.dim(i+1).size: uint;
+          start[i] = (locDom.dim(i+1).low - A.domain.dim(i+1).low): uint;
+        }
 
-      H5Dclose(dataset);
-      H5Sclose(sid);
-      H5Fclose(fid);
+        ret = H5Sselect_hyperslab(fileDataspace, H5S_SELECT_SET, start,
+                                  stride, count, nil);
+        if ret == FAIL then
+          halt("failed to select hyperslab");
+
+        // create a memory dataspace
+        var memDataspace = H5Screate_simple(A.rank, count, nil);
+        if memDataspace == FAIL then
+          halt("failed to create memory dataspace");
+
+        // set up the transfer properties list
+        var xferPlist = H5Pcreate(H5P_DATASET_XFER);
+        if xferPlist == FAIL then
+          halt("failed to create transfer properties list");
+
+        ret = H5Pset_dxpl_mpio(xferPlist, H5FD_MPIO_COLLECTIVE);
+        if ret == FAIL then
+          halt("failed to set data transfer property list");
+
+        // write data
+        ret = H5Dwrite(dataset, hdf5Type, memDataspace, fileDataspace,
+                       xferPlist, c_ptrTo(A._value.myLocArr!.myElems));
+        if ret == FAIL then
+          halt("failed writing array data to file");
+
+        // release temporary handles
+        H5Sclose(fileDataspace);
+        H5Sclose(memDataspace);
+        H5Pclose(xferPlist);
+
+        H5Dclose(dataset);
+        H5Sclose(sid);
+        H5Fclose(fid);
+      }
     }
-  }
-
```

Comment:
I think it would be good to restore this whitespace line

Reply:
Done, thanks!